### ISSUE TITLE
model predicts equal weights in portfolio

### ISSUE THREAD

**Issue by Amarchuk** [ZhengyaoJiang/PGPortfolio#87]
Hi there, 

I try to apply you project for forex data. While it is obviously different from crypto market, the differences are not that big and can be neglected or solved. Interesting, that all experiments with different assets (USD-based portfolio with 10 pairs for example) produce the models, which predicts near the same weight for all pairs in portfolio on every step. I interpreted this like the model can't generalize and can't prefer one pair to another.In other words every trained agent show behaviour similar to CRP.

Generally speaking, my question is - do you have some clues/ideas why this might happen? I think about number of steps (I don't use millions - just try 15k and 80k) as an obvious candidate along with fee can be big enough (I try to shrink it size without luck) or the situation where prices itself act in unison  (check that, not all pairs correlate with each other). Thanks

**Comment by DexHunter** [comment]
>  do you have some clues/ideas why this might happen?

There are many reasons this might happen. Like configuration of hyperparameters, technical bugs, etc. I am not sure what's the specific problem in your case.

> I try to shrink it size without luck

What do you mean by `shrink its size`? Are you referring to reduce the commission fee or steps?

> not all pairs correlate with each other

The assets in EIIE topology are actually independent. So I don't think that's the reason for uniform behavior.

**Comment by Amarchuk** [comment]
@DexHunter Yes, I try to reduce the commission fee as possible cause of such behaviour. 

So you doesn't experienced such equally-distributed model in any of your experiments? Interesting.

**Comment by Amarchuk** [comment]
@kumkee Thanks, I read original issue and found it very informative, but not sure how batch normalization can help in my situation.

**Comment by DexHunter** [comment]
>  So you doesn't experienced such equally-distributed model in any of your experiments?

Not really, the PVM initialization is uniform, so it seems to me the uniform behavior is like the agent didn't learn anything.

**Comment by ALevitskyy** [comment]
Got the same problem after training on ForEx data from histdata.com 

@Amarchuk Did you manage to figure out what`s happening?

**Comment by ALevitskyy** [comment]
After playing around with parameters, the model seems to converge to full cash and no trading after enough iterations. Seems like a good case for weak version of efficient market hypothesis or for my lack of skills
